test/btoken/TestBErc20_2.spec.ts

```diff
@@ -0,0 +1,552 @@
+
+      // USDT
+      bUSDT = await engine.deployNewBErc20("cUSDT");
+      bUSDT_addr = bUSDT.address;
+
+      USDT_addr = compoundUtil.getTokens("USDT");
+      USDT = await Erc20Detailed.at(USDT_addr);
+
+      cUSDT_addr = compoundUtil.getContracts("cUSDT");
+      cUSDT = await CErc20.at(cUSDT_addr);
+
+      // ETH:: deploy BEther
+      bETH = await engine.deployNewBEther();
+      bETH_addr = bETH.address;
+
+      cETH_addr = compoundUtil.getContracts("cETH");
+      cETH = await CErc20.at(cETH_addr);
+
+      expect((await ZRX.balanceOf(a.deployer)).gt(ZERO)).to.be.equal(true);
+      expect((await BAT.balanceOf(a.deployer)).gt(ZERO)).to.be.equal(true);
+
+      await ZRX.transfer(a.user1, ONE_THOUSAND_ZRX, { from: a.deployer });
+      expect(await ZRX.balanceOf(a.user1)).to.be.bignumber.equal(ONE_THOUSAND_ZRX);
+
+      await ZRX.transfer(a.user2, ONE_THOUSAND_ZRX, { from: a.deployer });
+      expect(await ZRX.balanceOf(a.user2)).to.be.bignumber.equal(ONE_THOUSAND_ZRX);
+
+      await BAT.transfer(a.user2, ONE_THOUSAND_BAT, { from: a.deployer });
+      expect(await BAT.balanceOf(a.user2)).to.be.bignumber.equal(ONE_THOUSAND_BAT);
+
+      await USDT.transfer(a.user4, ONE_THOUSAND_USDT, { from: a.deployer });
+      expect(await USDT.balanceOf(a.user4)).to.be.bignumber.equal(ONE_THOUSAND_USDT);
+
+      // NOTICE: Fix the Price oracle issue on Compound deployment
+      await comptroller._setPriceOracle(compoundUtil.getContracts("PriceOracle"));
+    });
+
+    describe("BErc20.borrow()", async () => {
+      let avatar1: string;
+      let avatar2: string;
+      let avatar3: string;
+      let avatar4: string;
+
+      beforeEach(async () => {
+        // user1 deposit ZRX
+        await ZRX.approve(bZRX_addr, ONE_THOUSAND_ZRX, { from: a.user1 });
+        await bZRX.mint(ONE_THOUSAND_ZRX, { from: a.user1 });
+        avatar1 = await bProtocol.registry.avatarOf(a.user1);
+
+        // user2 deposit BAT
+        await BAT.approve(bBAT_addr, ONE_THOUSAND_BAT, { from: a.user2 });
+        await bBAT.mint(ONE_THOUSAND_BAT, { from: a.user2 });
+        avatar2 = await bProtocol.registry.avatarOf(a.user2);
+
+        // user3 deposit ETH
+        await bETH.mint({ from: a.user3, value: TEN_ETH });
+        avatar3 = await bProtocol.registry.avatarOf(a.user3);
+
+        // user4 deposit USDT
+        await USDT.approve(bUSDT_addr, ONE_THOUSAND_USDT, { from: a.user4 });
+        await bUSDT.mint(ONE_THOUSAND_USDT, { from: a.user4 });
+        avatar4 = await bProtocol.registry.avatarOf(a.user4);
+      });
+
+      it("should borrow BAT", async () => {
+        // user1 borrows BAT, he has ZRX collateral
+        expect(await BAT.balanceOf(a.user1)).to.be.bignumber.equal(ZERO);
+
+        const err = await bBAT.borrow.call(ONE_BAT, { from: a.user1 });
+        expect(err).to.be.bignumber.equal(ZERO);
+        await bBAT.borrow(ONE_BAT, { from: a.user1 });
+
+        expect(await BAT.balanceOf(a.user1)).to.be.bignumber.equal(ONE_BAT);
```

Comment:
need to check that borrow balance current is as expected (though technically you already do it before)